onnx_trt_backend.cpp

```diff
@@ -539,91 +549,278 @@ onnxReleaseBackendID(onnxBackendID backendID) {
+          BACKEND_EXTENSIONS, strlen(BACKEND_EXTENSIONS));
+      case ONNXIFI_BACKEND_DEVICE:
+      {
+        cudaDeviceProp deviceProperties = { 0 };
+        cudaError_t cudaError =
+          cudaGetDeviceProperties(&deviceProperties, cudaDeviceId);
+        switch (cudaError) {
+          case cudaSuccess:
+            break;
+          case cudaErrorInvalidDevice:
+            return ONNXIFI_STATUS_INVALID_ID;
+          default:
+            return ONNXIFI_STATUS_INTERNAL_ERROR;
+        }
+        return setStringInfo(infoValue, infoValueSize,
+          deviceProperties.name,
+          strnlen(deviceProperties.name, sizeof(deviceProperties.name)));
       }
-      SET_UINT64(uint64_t(total));
-      break;
-    }
-    // TODO: Dummy numbers below
-    case ONNXIFI_BACKEND_MAX_GRAPH_SIZE:
-      SET_UINT64(1000000UL);
-      break;
-    case ONNXIFI_BACKEND_MAX_GRAPH_COUNT:
-      SET_UINT64(1UL);
-      break;
-    case ONNXIFI_BACKEND_MACS_FP32:
-      SET_UINT64(0UL);
-      break;
-    case ONNXIFI_BACKEND_MACS_FP16:
-      SET_UINT64(0UL);
-      break;
-    case ONNXIFI_BACKEND_MEMORY_BANDWIDTH:
-      SET_UINT64(0UL);
-      break;
-    case ONNXIFI_BACKEND_CPU_MEMORY_READ_BANDWIDTH:
-      SET_UINT64(0UL);
-      break;
-    case ONNXIFI_BACKEND_CPU_MEMORY_WRITE_BANDWIDTH:
-      SET_UINT64(0UL);
-      break;
-    default:
-      return ONNXIFI_STATUS_UNSUPPORTED_ATTRIBUTE;
+      case ONNXIFI_BACKEND_DEVICE_TYPE:
+        return setUIntInfo(infoValue, infoValueSize,
+          ONNXIFI_DEVICE_TYPE_GPU);
+      case ONNXIFI_BACKEND_ONNX_IR_VERSION:
+        return setStringInfo(infoValue, infoValueSize,
+          BACKEND_IR_VERSION, strlen(BACKEND_IR_VERSION));
+      case ONNXIFI_BACKEND_OPSET_VERSION:
+        return setStringInfo(infoValue, infoValueSize,
+          BACKEND_OPSET_VERSION, strlen(BACKEND_OPSET_VERSION));
+      case ONNXIFI_BACKEND_CAPABILITIES:
+        return setUIntInfo(infoValue, infoValueSize, 0);
+      case ONNXIFI_BACKEND_INIT_PROPERTIES:
+        return setUIntInfo(infoValue, infoValueSize, 0);
+      case ONNXIFI_BACKEND_MEMORY_TYPES:
+        return setUIntInfo(infoValue, infoValueSize,
+          ONNXIFI_MEMORY_TYPE_CPU | ONNXIFI_MEMORY_TYPE_CUDA_BUFFER);
+      case ONNXIFI_BACKEND_GRAPH_INIT_PROPERTIES:
+        return setUIntInfo(infoValue, infoValueSize, 0);
+      case ONNXIFI_BACKEND_SYNCHRONIZATION_TYPES:
+        return setUIntInfo(infoValue, infoValueSize,
+          ONNXIFI_SYNCHRONIZATION_EVENT);
+      case ONNXIFI_BACKEND_MAX_GRAPH_COUNT:
+        return setUIntInfo(infoValue, infoValueSize,
+          std::numeric_limits<uint64_t>::max());
+      case ONNXIFI_BACKEND_CPU_MEMORY_READ_BANDWIDTH:
+      case ONNXIFI_BACKEND_CPU_MEMORY_WRITE_BANDWIDTH:
+        /* Assume PCI Express 3.0 x16 */
+        return setUIntInfo(infoValue, infoValueSize,
+          UINT64_C(16519104985));
+      case ONNXIFI_BACKEND_MEMORY_SIZE:
+      case ONNXIFI_BACKEND_GRAPH_SIZE:
```

Comment:
`ONNXIFI_BACKEND_GRAPH_SIZE` is no longer defined in onnxifi.h. Please replace it globally.